### ISSUE TITLE
abortNavigation() does not abort navigation when `ssr: false`

### ISSUE THREAD

**Issue by StefanCardnell** [nuxt/framework#6849]
### Environment

- Operating System: `Linux`
- Node Version:     `v18.5.0`
- Nuxt Version:     `3.0.0-rc.8`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `ssr`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

1) Use `ssr: false` in `nuxt.config.ts`
2) Create `middleware/auth.global.ts` and put the following simple config:

```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
    console.log("aborting!");
    return abortNavigation();
})
```

3) `npm run dev`

### Describe the bug

The page should not be routed to. A 500 error should appear.

However instead, the page is routed to like normal.

Without specifying `ssr: false`, a 500 error appears like normal.

**Comment by danielroe** [comment]
When `ssr: false` nuxt route middleware and plugins do not run on the server.

**Comment by StefanCardnell** [comment]
Hi @danielroe. This is a client-only based SPA. When disabling `ssr`, client-side route middleware defined in `middleware` will still run. However `abortNavigation()` is doing nothing - it should result in a 500 error or so anything that doesn't route to the actual page?

I've noticed that if I put a string in the `abortNavigation`, the bug seems to be fixed.

```
export default defineNuxtRouteMiddleware(async (to, from) => {
    console.log("aborting!");
    return abortNavigation("Aborted!");
})
```

I noticed there is another open issue with the same problem:

https://github.com/nuxt/framework/issues/5467
